backend/git_real/tests/test_api_views.py

```diff
@@ -1,13 +1,68 @@
 from rest_framework.test import APITestCase
 from test_mixins import APITestCaseMixin
+from .factories import RepositoryFactory
+from .utils import get_body_and_headers
+from django.urls import reverse
+from django.utils import timezone
+import dateutil.parser
+import mock
 
-
-from . import factories
+from git_real import views
+from git_real.models import Push
+from git_real.permissions import IsWebhookSignatureOk
+from git_real.helpers import github_timestamp_int_to_tz_aware_datetime
 
 
 class RepositoryViewsetTests(APITestCase, APITestCaseMixin):
     LIST_URL_NAME = "repository-list"
     SUPPRESS_TEST_POST_TO_CREATE = True
 
     def verbose_instance_factory(self):
-        return factories.RepositoryFactory(archived=True)
+        return RepositoryFactory(archived=True)
+
+
+class PushEventTests(APITestCase):
+    @mock.patch.object(IsWebhookSignatureOk, "has_permission")
+    def test_push_event_is_retrieved_from_api_end_point(self, has_permission):
+        has_permission.return_value = True
+        url = reverse(views.github_webhook)
+        body, headers = get_body_and_headers("push")
+        repo = RepositoryFactory(full_name=body["repository"]["full_name"])
+
+        repository = body["repository"]
+        head_commit = body["head_commit"]
+        author_github_name = head_commit["author"]["username"]
+        committer_github_name = head_commit["committer"]["username"]
+        pusher_username = body["pusher"]["name"]
+        message = head_commit["message"]
+        head_commit_url = head_commit["url"]
+        commit_timestamp = dateutil.parser.isoparse(head_commit["timestamp"])
+        ref = body["ref"]
+        pushed_at_time = github_timestamp_int_to_tz_aware_datetime(
+            int(repository["pushed_at"])
+        )
+
+        self.client.post(url, format="json", data=body, extra=headers)
+        self.assertEqual(Push.objects.count(), 1)
+        push = Push.objects.first()
```

Comment:
This seems very complicated. Can't you use a factory to make a Push instance? Just `push = PushFactory()`

Here you are testing that the webhook works. We already tested the webhook elsewhere.  And it's a totally different api endpoint.